compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -633,28 +635,43 @@ class ComposeScene internal constructor(
             // If pressOwner is under focusedOwner, hover state must be updated
             owner = null
         }
+        if (processHover(event, owner)) {
+            return
+        }
+        owner?.processPointerInput(
+            event.copy(eventType = PointerEventType.Move)
+        )
+    }
 
+    /**
+     * Updates hover state and generates [PointerEventType.Enter] and [PointerEventType.Exit]
+     * events. Returns true if [event] is consumed.
+     */
+    private fun processHover(event: PointerInputEvent, owner: SkiaBasedOwner?): Boolean {
+        if (!event.pointers.fastAny { it.type == PointerType.Mouse }) {
+            // Track hover only for mouse
+            return false
+        }
         // Cases:
         // - move from outside to the window (owner != null, lastMoveOwner == null): Enter
         // - move from the window to outside (owner == null, lastMoveOwner != null): Exit
         // - move from one point of the window to another (owner == lastMoveOwner): Move
         // - move from one popup to another (owner != lastMoveOwner): [Popup 1] Exit, [Popup 2] Enter
-
-        if (owner != lastMoveOwner && event.pointers.fastAny { it.type == PointerType.Mouse }) {
-            lastMoveOwner?.processPointerInput(
-                event.copy(eventType = PointerEventType.Exit),
-                isInBounds = false
-            )
-            owner?.processPointerInput(
-                event.copy(eventType = PointerEventType.Enter)
-            )
-        } else {
-            owner?.processPointerInput(
-                event.copy(eventType = PointerEventType.Move)
-            )
+        if (owner == lastHoverOwner) {
+            // Owner wasn't changed
+            return false
```

Comment:
Maybe put this test first, since it's faster than iterating over pointers.

Reply:
It will be logically incorrect because this check makes sense only for mouse.

For speed comparison: it won't be differ because it should end on first iteration - mouse always have only one pointer

Reply:
The `==` test was first before, and it doesn't have any side effects, so I don't know...